packages/editor-ui/src/permissions.ts

```diff
@@ -13,13 +13,16 @@ export type PermissionsMap<T> = {
 	[K in ExtractAfterColon<T>]: boolean;
 };
 
-const mapScopesToPermissions = (scopes: Scope[], scopeSet: Set<Scope>): PermissionsMap<Scope> =>
+const mapScopesToPermissions = <T extends Scope>(
+	scopes: T[],
+	scopeSet: Set<T>,
+): PermissionsMap<T> =>
 	scopes.reduce(
```

Comment:
Not sure if it works, but `reduce` accepts a generic `scopes.reduce<PermissionsMap<T>>`. To avoid type casting the accumulator. Ignore if it doesn't work.

Reply:
In most of the cases, it never worked for me; I don't know why.
Here I get `Argument of type '{}' is not assignable to parameter of type 'PermissionsMap<T>'.ts(2345)`
It would work only if would use `Object.create(null)` as the seed instead of the object literal (`{}`)

Reply:
Ignore it then, not a big deal.